code/modules/mob/living/ventcrawling.dm

```diff
@@ -97,7 +97,7 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, typecacheof(list(
 			A.pipe_vision_img = image(A, A.loc, layer = ABOVE_HUD_LAYER, dir = A.dir)
 			A.pipe_vision_img.plane = ABOVE_HUD_PLANE
 		pipes_shown += A.pipe_vision_img
-		if(client)
+		if(client && A.loc.x >= starting_machine.loc.x - 8 && A.loc.x <= starting_machine.loc.x + 8 && A.loc.y >= starting_machine.loc.y - 8 && A.loc.y <= starting_machine.loc.y + 8)
```

Comment:
```suggestion
		if(client && in_view_range(client.mob, A))
```

Reply:
that said, i'm not sure if anything in this for is really necessary if there isn't a client to be showing the pipes to in the first place. could probably nest `for(var/X in totalMembers)` under a `if(client`) check to save some processing to start with.

Reply:
Oh cool, I was looking for a proc like that. Tested, works well. I also moved the if(client) out of the for loop, which got me thinking. 
We don't really need to check if a pipe has an image set if we never see it, so that is also inside the if now. Not moving pipes_shown the first time was an oversight, as it is used to remove images in the remove_ventcrawl() proc.